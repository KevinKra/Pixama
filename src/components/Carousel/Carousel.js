import React from "react";

export default function Carousel() {
  return (
    <div>
      <p>Carousel</p>
    </div>
  );
}
