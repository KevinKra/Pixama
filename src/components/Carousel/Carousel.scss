@import "../../_styles//variables.scss";

.Carousel {
  border: 1px solid orange;
  margin-top: 40px;
  position: relative;
  z-index: -1;
  overflow: hidden;
  h2 {
    margin: 0.1rem;
    padding-left: 1.7rem;
  }
}

.movies-container {
  position: relative;
  z-index: 1;
  display: flex;
  overflow-y: none;
  width: 3800px;
  border: 1px solid red;
  transition: 1s transform ease-in-out;
  transform: translateX(0px);
  padding: 2rem 55px;
}

.backwards-btn,
.forwards-btn {
  z-index: 3;
  position: absolute;
  color: white;
  border: none;
  // border: 1px solid black;
  width: 5%;
  height: 100%;
  background: transparent;
  &:focus {
    outline: none;
  }
}

.forwards-btn {
  left: 0;
  background: linear-gradient(
    90deg,
    $primary-color 0%,
    $primary-color 19%,
    rgba(27, 27, 27, 0.65) 64%,
    rgba(27, 27, 27, 0.56) 80%,
    rgba(27, 27, 27, 0) 98%
  );
}

.backwards-btn {
  right: 0;
  background: rgb(0, 0, 0);
  background: linear-gradient(
    270deg,
    $primary-color 0%,
    $primary-color 19%,
    rgba(27, 27, 27, 0.65) 64%,
    rgba(27, 27, 27, 0.56) 80%,
    rgba(27, 27, 27, 0) 98%
  );
}

.backwards-btn {
}

.movies-container::-webkit-scrollbar {
  width: 0 !important;
  overflow: -moz-scrollbars-none;
  -ms-overflow-style: none;
}
