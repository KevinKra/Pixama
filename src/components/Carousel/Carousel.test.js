import { Carousel } from "./Carousel";
import React from "react";
import { shallow } from "enzyme";

describe("Carousel", () => {
  let wrapper;
  let instance;
  const mockPopMovies = [{ title: "movie 1" }, { title: "movie 2" }];
  const mockRomMovies = [{ title: "movie 1" }, { title: "movie 2" }];
  const mockFavs = [{ title: "movie 1" }, { title: "movie 2" }];

  beforeEach(() => {
    wrapper = shallow(<Carousel title={"Favorites"} genre={"favorites"} popularMovies={mockPopMovies} romanceMovies={mockRomMovies} favorites={mockFavs} />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it ('should have default state', () => {
    expect (wrapper.state('movies')).toEqual(mockFavs);
  });

  it ('should not render movie cards if data hasnt yet been fetched', () => {
    wrapper = shallow(
      <Carousel
        title={"Favorites"}
        genre={"favorites"}
        popularMovies={mockPopMovies}
        romanceMovies={mockRomMovies}
        favorites={[]}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  
});
