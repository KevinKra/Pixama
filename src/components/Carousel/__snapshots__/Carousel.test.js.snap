// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match the snapshot 1`] = `
<div
  className="Carousel"
>
  <h2>
    Favorites
  </h2>
  <button
    className="forwards-btn"
    onClick={[Function]}
  >
    <span>
      ⏪
    </span>
  </button>
  <button
    className="backwards-btn"
    onClick={[Function]}
  >
    <span>
      ⏩
    </span>
  </button>
  <div
    className="movies-container"
    style={
      Object {
        "transform": "translateX(-25%)",
      }
    }
  >
    <Connect(withRouter(MovieCard)) />
    <Connect(withRouter(MovieCard)) />
  </div>
</div>
`;

exports[`Carousel should not render movie cards if data hasnt yet been fetched 1`] = `
<div
  className="Carousel"
>
  <h2>
    Favorites
  </h2>
  <button
    className="forwards-btn"
    onClick={[Function]}
  >
    <span>
      ⏪
    </span>
  </button>
  <button
    className="backwards-btn"
    onClick={[Function]}
  >
    <span>
      ⏩
    </span>
  </button>
  <div
    className="movies-container"
    style={
      Object {
        "transform": "translateX(-25%)",
      }
    }
  />
</div>
`;
